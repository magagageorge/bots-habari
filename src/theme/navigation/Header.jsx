import React from 'react';
import logo from '../../assets/img/logo/logo.png';
import {
    Link, withRouter
} from "react-router-dom";

let header_area;
let sticky;
class Header extends React.Component {
    constructor(props) {
        super(props);
        this.app_header_area = React.createRef();
    }

    componentDidUpdate(prevProps) {
        if (this.props.location !== prevProps.location) {
            this.onRouteChanged();
        }
    }

    onRouteChanged() {
        //console.log("ROUTE CHANGED");
    }

    componentDidMount() {
        window.addEventListener('scroll', this.handleSticky);
    }

    componentWillUnmount() {
        window.removeEventListener('scroll', this.handleSticky);
    }

    /** This functions does the magic or either adding or removing css sticky class to the top Navigation Menu */
    handleSticky = () => {
        header_area = this.refs.app_header_area;
        sticky = header_area.offsetTop; /* Get the offset position of the Top Menu Header */
        if (window.pageYOffset > sticky) {
            header_area.classList.add("sticky");
        } else {
            header_area.classList.remove("sticky");
        }
    }

    render() {
        var path = window.location.pathname;
        var path_routes = path.split('/');
        var header_name = (path_routes[1] != '') ? ((path_routes[1] == 'articles') ? 'Article' : 'Page') : '';
        return (
            <header>
                {/* Header Start */}
                <div className="header-area bg-white" ref="app_header_area">
                    <div className="main-header">
                        <div className="header-mid d-none d-md-block">
                            <div className="container fluid">
                                <div className="row mx-0">
                                    {/* Logo */}
                                    <div className="col-12 d-flex justify-content-center">
                                        <img src={logo} className="img-fluid" alt="" />
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="header-bottom header-sticky">
                            <div className="container fluid px-0">
                                <div className="row mx-0">
                                    <div className="col-2 d-block d-md-none d-lg-none d-xl-none">
                                        {path === '/' && <div data-toggle="collapse" data-target="#navbarContent" aria-controls="navbars" aria-expanded="false" aria-label="Toggle navigation" className="text-primary w-100 h-100 d-flex justify-content-center align-items-center navbar-toggler" style={{ fontSize: '130%' }}>
                                            <span className="fa fa-bars"></span>
                                        </div>}
                                        {path != '/' && <div className="text-primary w-100 h-100 d-flex justify-content-center align-items-center navbar-toggler" style={{ fontSize: '130%' }}>
                                            <Link to="/" className="d-block text-primary" ><span className="fas fa-arrow-left"></span></Link>
                                        </div>}
                                    </div>
                                    <div className="col-8 col-lg-10 col-md-12 header-flex">
                                        <div className="row mx-0">
                                            <div className="col-12 d-block d-md-none d-lg-none d-xl-none px-0">
                                                {header_name !== '' && <div className="d-flex justify-content-center align-items-center w-100 py-3">{header_name}</div>}
                                                {header_name === '' && <Link to="/" className="w-100"><img src={logo} alt="" className="img-fluid my-1" /></Link>}
                                            </div>
                                            <div className="col-12 d-none d-md-block d-lg-block d-xl-block px-0">
                                                {/* Main-menu */}
                                                <div className="main-menu d-none d-md-block">
                                                    <nav>
                                                        <ul id="navigation">
                                                            <li><Link to="/">Home</Link></li>
                                                            <li><Link to="/page">Page 1</Link></li>
                                                            <li><Link to="/page">Page 2</Link></li>
                                                            <li><Link to="/page">Page 3</Link></li>
                                                            <li><Link to="/page">Page 4</Link></li>
                                                        </ul>
                                                    </nav>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                    <div className="col-2 col-lg-2 col-md-4 d-flex align-items-center justify-content-end">
                                        <div className="header-right-btn f-right">
                                            <i className="fas fa-search special-tag"></i>
                                            <div className="search-box">
                                                <form>
                                                    <input type="text" placeholder="Search" />
                                                </form>
                                            </div>
                                        </div>
                                    </div>
                                    {/* Mobile Menu */}
                                    <div className="col-12 d-block d-md-none d-lg-none d-xl-none px-0">
                                        <div className="position-absolute" style={{ left: '0px', right: '0px' }}>
                                            <nav className="navbar navbar-expand-lg navbar-light bg-white p-0 shadow-sm">
                                                <div id="navbarContent" className="collapse navbar-collapse mobile-menu">
                                                    <ul className="navbar-nav mx-auto px-2">
                                                        <li className="nav-item"><Link to="/" data-toggle="collapse" data-target="#navbarContent" className="nav-link">Home</Link></li>
                                                        <li className="nav-item"><Link to="/page" data-toggle="collapse" data-target="#navbarContent" className="nav-link">Page 1</Link></li>
                                                        <li className="nav-item"><Link to="/page" data-toggle="collapse" data-target="#navbarContent" className="nav-link">Page 2</Link></li>
                                                        <li className="nav-item"><Link to="/page" data-toggle="collapse" data-target="#navbarContent" className="nav-link">Page 3</Link></li>
                                                        <li className="nav-item"><Link to="/page" data-toggle="collapse" data-target="#navbarContent" className="nav-link">Page 4</Link></li>
                                                    </ul>
                                                </div>
                                            </nav>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                {/* Header End */}
            </header>

        )
    }
}

export default withRouter(Header);