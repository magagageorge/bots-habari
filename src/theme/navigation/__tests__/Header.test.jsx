
// Auto-generated do not edit


/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import { BrowserRouter } from 'react-router-dom';


describe('theme/navigation/Header test', () => {
  it('theme/navigation/Header should match snapshot', () => {
    const component = renderer.create(<BrowserRouter>
      <Header />
    </BrowserRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
