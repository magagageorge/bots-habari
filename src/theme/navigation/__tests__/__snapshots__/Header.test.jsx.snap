// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme/navigation/Header test theme/navigation/Header should match snapshot 1`] = `
<header>
  <div
    className="header-area bg-white"
  >
    <div
      className="main-header"
    >
      <div
        className="header-mid d-none d-md-block"
      >
        <div
          className="container fluid"
        >
          <div
            className="row mx-0"
          >
            <div
              className="col-12 d-flex justify-content-center"
            >
              <img
                alt=""
                className="img-fluid"
                src="logo.png"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="header-bottom header-sticky"
      >
        <div
          className="container fluid px-0"
        >
          <div
            className="row mx-0"
          >
            <div
              className="col-2 d-block d-md-none d-lg-none d-xl-none"
            >
              <div
                aria-controls="navbars"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="text-primary w-100 h-100 d-flex justify-content-center align-items-center navbar-toggler"
                data-target="#navbarContent"
                data-toggle="collapse"
                style={
                  Object {
                    "fontSize": "130%",
                  }
                }
              >
                <span
                  className="fa fa-bars"
                />
              </div>
            </div>
            <div
              className="col-8 col-lg-10 col-md-12 header-flex"
            >
              <div
                className="row mx-0"
              >
                <div
                  className="col-12 d-block d-md-none d-lg-none d-xl-none px-0"
                >
                  <a
                    className="w-100"
                    href="/"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      className="img-fluid my-1"
                      src="logo.png"
                    />
                  </a>
                </div>
                <div
                  className="col-12 d-none d-md-block d-lg-block d-xl-block px-0"
                >
                  <div
                    className="main-menu d-none d-md-block"
                  >
                    <nav>
                      <ul
                        id="navigation"
                      >
                        <li>
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            Home
                          </a>
                        </li>
                        <li>
                          <a
                            href="/page"
                            onClick={[Function]}
                          >
                            Page 1
                          </a>
                        </li>
                        <li>
                          <a
                            href="/page"
                            onClick={[Function]}
                          >
                            Page 2
                          </a>
                        </li>
                        <li>
                          <a
                            href="/page"
                            onClick={[Function]}
                          >
                            Page 3
                          </a>
                        </li>
                        <li>
                          <a
                            href="/page"
                            onClick={[Function]}
                          >
                            Page 4
                          </a>
                        </li>
                      </ul>
                    </nav>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-2 col-lg-2 col-md-4 d-flex align-items-center justify-content-end"
            >
              <div
                className="header-right-btn f-right"
              >
                <i
                  className="fas fa-search special-tag"
                />
                <div
                  className="search-box"
                >
                  <form>
                    <input
                      placeholder="Search"
                      type="text"
                    />
                  </form>
                </div>
              </div>
            </div>
            <div
              className="col-12 d-block d-md-none d-lg-none d-xl-none px-0"
            >
              <div
                className="position-absolute"
                style={
                  Object {
                    "left": "0px",
                    "right": "0px",
                  }
                }
              >
                <nav
                  className="navbar navbar-expand-lg navbar-light bg-white p-0 shadow-sm"
                >
                  <div
                    className="collapse navbar-collapse mobile-menu"
                    id="navbarContent"
                  >
                    <ul
                      className="navbar-nav mx-auto px-2"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-target="#navbarContent"
                          data-toggle="collapse"
                          href="/"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-target="#navbarContent"
                          data-toggle="collapse"
                          href="/page"
                          onClick={[Function]}
                        >
                          Page 1
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-target="#navbarContent"
                          data-toggle="collapse"
                          href="/page"
                          onClick={[Function]}
                        >
                          Page 2
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-target="#navbarContent"
                          data-toggle="collapse"
                          href="/page"
                          onClick={[Function]}
                        >
                          Page 3
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          data-target="#navbarContent"
                          data-toggle="collapse"
                          href="/page"
                          onClick={[Function]}
                        >
                          Page 4
                        </a>
                      </li>
                    </ul>
                  </div>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
